### ISSUE TITLE
Camera light still on after local tracks are stopped

### ISSUE THREAD

**Issue by Roshan931** [twilio/twilio-video.js#272]
- [x] I have verified that the issue occurs with the latest twilio-video.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md).
 - [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-video.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [x] I verified that the [Quickstart](https://github.com/twilio/video-quickstart-js) application works in my environment.

**Code to reproduce the issue:**

Connecting to Twilio and displaying the local and remote tracks works perfectly. However, when I try to turn off the camera or to completely disconnect from the room, the camera light is still on and the permission for camera is still allowed.

Everything works fine in the quickstart project and I am using the same code from there. I don't know if it has to do something with the Angular environment. 

```js
// Creating local tracks
this.tracks = await Twilio.Video.createLocalTracks({ audio: true, video: { width: 640 }})

// I pass the localTracks created previously and a uuid for name
this.room = await Twilio.Video.connect(token, { name: name, tracks: this.tracks })

// Here is the code where I disable the video. The video is not showing anymore but the camera light is still on
this.room.localParticipant.videoTracks.forEach(track => {
      track.disable()
})

// Here is the code when the room is disconnected. I used similar code as in the quickstart-video. This is in the on('disconnected') callback 
if (this.tracks) {
    this.tracks.forEach(track => {
        track.stop()
    })
}

// I also tried invoking stop(), detach() and/or removeTrack() on the localParticipant tracks
let tracks = Array.from(this.room.localParticipant.tracks.values())
tracks.forEach((track: any) => {
    track.detach().forEach(detachedElement => {
        detachedElement.remove()
    })
})

this.room.participants.forEach(participant => {
    let pTracks = Array.from(participant.tracks.values())
    pTracks.forEach((track: any) => {
        track.detach().forEach(detachedElement => {
            detachedElement.remove()
        })
    })
})

this.room = null
```

**Expected behavior:**

Remove the preview of local tracks and turn off camera when disabling video or when the room is disconnected.

**Actual behavior:**

Removes the preview but the camera light is still on.

**Logs**

The logs clearly show that the LocalVideoTrack is being disabled and ended. I also looked into the LocalVideoTrack object after it is being stopped and it returns true for isStopped and false for isEnabled.

![screen shot 2018-03-02 at 11 27 07 am](https://user-images.githubusercontent.com/5818108/36896017-432caa5e-1e11-11e8-91ed-9c93d24f812b.png)

**Software versions:**

 - [x] Browser(s): Safari 11.0.3, Chrome 64.0.3282.186
 - [x] Operating System: macOS High Sierra
 - [x] twilio-video.js: //media.twiliocdn.com/sdk/js/video/v1/twilio-video.min.js
 - [x] Third-party libraries (e.g., Angular, React, etc.): Angular 5

**Comment by markandrus** [comment]
Hi, I believe the issue is that `createLocalTracks` returns a Promise, so you need to `await` or call `then` on the Promise in order to get the `tracks` out.

**Comment by Roshan931** [comment]
Hi @markandrus. Sorry, I forgot to add the await when I created the issue. I am awaiting it in my code. I updated the comment.

**Comment by markandrus** [comment]
Please share the full logs from the page so we can identify where each LocalAudioTack/LocalVideoTrack is created and then where each is stopped. Please also pass `logLevel: 'debug'` in both `connect` and `createLocalTracks`. Can you also

* Confirm you only call `createLocalTracks` where you mentioned?
* Confirm you only call `connect` where you mentioned?
* Confirm you perform no other operations that invoke `getUserMedia`?
* What is `this`?
* Are you perhaps overwriting `this.tracks` and losing references to LocalVideoTracks?

LocalVideoTrack#stop and LocalAudioTrack#stop just delegate to the underlying MediaStreamTrack's `stop`. It's equivalent to doing

```js
localTrack.mediaStreamTrack.stop()
```

So if you call `stop` and `isStopped` returns `true`, you can be sure _those_ MediaStreamTracks are stopped (another way to check: `mediaStreamTrack.readyState === 'ended'`). However, so long as one video MediaStreamTrack on the page is not ended, the camera light will remain on.

**Comment by Roshan931** [comment]
While reading your comment and going through the logs I realized that I need to stop the tracks on one more place. I have a screen where I preview the local tracks before going into the call screen. So the `createLocalTracks` promise was called twice and the first time before switching the screen the tracks were not stopped.

I stopped the first tracks and now it is working. I thought that if I route to another angular component, the previous will destroy and the tracks will be stopped automatically. However, thank you very much for your help.
